File: app/src/main/java/com/roy/techpoint/annotation/java/BankService.java
package com.roy.techpoint.annotation.java;

import java.lang.reflect.Method;

public class BankService {

    @BankTransferMoney(maxMoney = 15000)
    public static void TransferMoney(double money) {
          System.out.println(processAnnotationMoney(money));
    }

    private static String processAnnotationMoney(double money) {
        try {
            Method transferMoney = BankService.class.getDeclaredMethod("TransferMoney", double.class);
            boolean annotationPresent = transferMoney.isAnnotationPresent(BankTransferMoney.class);
            if (annotationPresent) {
                BankTransferMoney annotation = transferMoney.getAnnotation(BankTransferMoney.class);
                double maxMoney = annotation.maxMoney();
                if (money > maxMoney) {
                    return "转账金额大于限额，转账失败";
                } else {
                    return "转账金额为:" + money + "，转账成功";
                }
            }
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        }
        return "转账处理失败";
    }

    public static void main(String[] args) {
        TransferMoney(20000);
    }
}

File: app/src/main/java/com/roy/techpoint/annotation/kotlin/BankService.kt
package com.roy.techpoint.annotation.kotlin

object BankService {
    @BankTransferMoney(maxMoney = 15000.00)
    fun transferMoney(money: Double) {
        println(processAnnotationMoney(money))
    }

    private fun processAnnotationMoney(money: Double): String {
        try {
            val transferMoney = BankService::class.java.getDeclaredMethod(
                "transferMoney",
                Double::class.javaPrimitiveType
            )
            val annotationPresent = transferMoney.isAnnotationPresent(BankTransferMoney::class.java)
            if (annotationPresent) {
                val annotation = transferMoney.getAnnotation(
                    BankTransferMoney::class.java
                )
                val maxMoney = annotation.maxMoney
                return if (money > maxMoney) {
                    "转账金额大于限额，转账失败"
                } else {
                    "转账金额为:$money，转账成功"
                }
            }
        } catch (e: NoSuchMethodException) {
            e.printStackTrace()
        }
        return "转账处理失败"
    }

    @JvmStatic
    fun main(args: Array<String>) {
        transferMoney(10000.0)
    }
}